src/api/operator/numpy/np_histogram_op.cc

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file np_histogram_op.cc
+ * \brief Implementation of the API of functions in src/operator/tensor/histogram.cc
+ */
+
+#include <mxnet/api_registry.h>
+#include <mxnet/runtime/packed_func.h>
+#include "../utils.h"
+#include "../../../operator/tensor/histogram-inl.h"
+
+namespace mxnet {
+
+MXNET_REGISTER_API("_npi.histogram")
+.set_body([](runtime::MXNetArgs args, runtime::MXNetRetValue* ret) {
+  using namespace runtime;
+  nnvm::NodeAttrs attrs;
+  const nnvm::Op* op = Op::Get("_npi_histogram");
+  op::HistogramParam param;
+  // parse bin_cnt
+  if (args[2].type_code() == kNull) {
+    param.bin_cnt = dmlc::nullopt;
+  } else {
+    param.bin_cnt = args[2].operator int();
+  }
+
+  // parse range
+  if (args[3].type_code() == kNull) {
+    param.range = dmlc::nullopt;
+  } else {
+    param.range = Tuple<double>(args[3].operator ObjectRef());
+  }
+
+  attrs.parsed = std::move(param);
+  attrs.op = op;
+  SetAttrDict<op::HistogramParam>(&attrs);
+
+  std::vector<NDArray*> inputs_vec;
+  int num_inputs = 0;
+
+  if (param.bin_cnt.has_value()) {
```

Comment:
`param` has been moved. We'd better check `args[2]` here.

Reply:
Done